File: 1620-check-if-array-pairs-are-divisible-by-k/solution.java
class Solution {
    public boolean canArrange(int[] arr, int k) {
        int[] count = new int[k];
        
        for (int num : arr) {
            int remainder = ((num % k) + k) % k;
            count[remainder]++;
        }

        if (count[0] % 2 != 0) {
            return false;
        }

        for (int i = 1; i < (k + 1) / 2; i++) {
            if (count[i] != count[k - i]) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5, 10, 6, 7, 8, 9};
        int k = 5;
        Solution solution = new Solution();
        boolean answer = solution.canArrange(arr, k);
        System.out.println(answer);
    }
}

File: 1620-check-if-array-pairs-are-divisible-by-k/solution.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        dictionary={}
        for i in range(len(arr)):
            modulus=(arr[i]%k)
            if modulus not in dictionary:
                dictionary[modulus] = 0
            dictionary[modulus]+=1
        for (key,val) in dictionary.items():
            if(key == 0):
                print(dictionary[key])
                if(dictionary[key]%2!=0):
                    return False
                else:
                    continue
            elif key == (k-key):
                if(dictionary[key]%2 !=0):
                    return False
                else:
                    continue
            elif k-key not in dictionary:
                return False
            elif(dictionary[key] != dictionary[k-key]):
                
                return False
        return True 
        



        
